src/transformers/models/idefics/configuration_idefics.py

```diff
@@ -163,22 +156,40 @@ def __init__(
         self.vision_intermediate_size = vision_intermediate_size
         self.vision_patch_size = vision_patch_size
 
-        if vision_config is None:
-            self.vision_config = {}
-            self.vision_config["hidden_size"] = vision_embed_dim
-            self.vision_config["image_size"] = vision_image_size
-            self.vision_config["num_attention_heads"] = num_attention_heads
-            self.vision_config["num_hidden_layers"] = num_hidden_layers
-            self.vision_config["intermediate_size"] = vision_intermediate_size
-            self.vision_config["patch_size"] = vision_patch_size
-        elif not isinstance(vision_config, dict):
-            raise ValueError("vision_config must be a dict")
-        else:
-            self.vision_config = vision_config
+        self.vision_config_dict = {}
+        self.vision_config_dict["hidden_size"] = vision_embed_dim
+        self.vision_config_dict["image_size"] = vision_image_size
+        self.vision_config_dict["num_attention_heads"] = vision_num_attention_heads
+        self.vision_config_dict["num_hidden_layers"] = vision_num_hidden_layers
+        self.vision_config_dict["intermediate_size"] = vision_intermediate_size
+        self.vision_config_dict["patch_size"] = vision_patch_size
```

Comment:
does it read better as this? I guess it somewhat breaks the one line per assignment pattern.

```suggestion
        self.vision_config_dict = dict(
            hidden_size=vision_embed_dim,
            image_size=vision_image_size,
            num_attention_heads=num_attention_heads,
            num_hidden_layers=num_hidden_layers,
            intermediate_size=vision_intermediate_size,
            patch_size=vision_patch_size,
        )
```

I wasn't sure.

But we should drop `vision_` prefix in `num_hidden_layers` and `num_attention_heads`